generators/csharp/codegen/src/FileGenerator.ts

```diff
@@ -12,10 +12,20 @@ export abstract class FileGenerator<
     constructor(protected readonly context: Context) {}
 
     public generate(): GeneratedFile {
-        this.context.logger.debug(`Generating ${this.getFilepath()}`);
+        if (this.shouldGenerate()) {
```

Comment:
@amckinney WDYT of this change?
I'm not forcing existing code to respect this new `shouldGenerate` method, so just logging it as a warning if it's called when not true.

This lets me remove the WireTest for custom pagination

Reply:
Interesting, I'm fine with this change since it's largely a no-op. 

Although it's unlikely to happen, the internal warning would be confusing to the user. I know it's more for us though.

Let's go with this for now, and we can always revisit later if it ends up causing confusion.